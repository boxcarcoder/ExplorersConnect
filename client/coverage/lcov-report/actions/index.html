
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for actions</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> actions</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">94.58% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>192/203</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">53.33% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>16/30</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>45/45</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">93.92% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>170/181</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="alert.js"><a href="alert.js.html">alert.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="auth.js"><a href="auth.js.html">auth.js</a></td>
	<td data-value="96.67" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 96%"></div><div class="cover-empty" style="width: 4%"></div></div>
	</td>
	<td data-value="96.67" class="pct high">96.67%</td>
	<td data-value="30" class="abs high">29/30</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="96.15" class="pct high">96.15%</td>
	<td data-value="26" class="abs high">25/26</td>
	</tr>

<tr>
	<td class="file high" data-value="post.js"><a href="post.js.html">post.js</a></td>
	<td data-value="90.63" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 90%"></div><div class="cover-empty" style="width: 10%"></div></div>
	</td>
	<td data-value="90.63" class="pct high">90.63%</td>
	<td data-value="64" class="abs high">58/64</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="12" class="abs medium">6/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="16" class="abs high">16/16</td>
	<td data-value="89.29" class="pct high">89.29%</td>
	<td data-value="56" class="abs high">50/56</td>
	</tr>

<tr>
	<td class="file high" data-value="profile.js"><a href="profile.js.html">profile.js</a></td>
	<td data-value="94.81" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 94%"></div><div class="cover-empty" style="width: 6%"></div></div>
	</td>
	<td data-value="94.81" class="pct high">94.81%</td>
	<td data-value="77" class="abs high">73/77</td>
	<td data-value="53.33" class="pct medium">53.33%</td>
	<td data-value="15" class="abs medium">8/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="18" class="abs high">18/18</td>
	<td data-value="94.2" class="pct high">94.2%</td>
	<td data-value="69" class="abs high">65/69</td>
	</tr>

<tr>
	<td class="file high" data-value="types.js"><a href="types.js.html">types.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="26" class="abs high">26/26</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="26" class="abs high">26/26</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Tue Oct 06 2020 19:01:54 GMT-0700 (Pacific Daylight Time)
            </div>
        </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    