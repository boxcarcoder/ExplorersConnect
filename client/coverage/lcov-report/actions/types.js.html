
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for actions/types.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> / <a href="index.html">actions</a> types.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>26/26</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>26/26</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const SET_ALERT = 'SET_ALERT';
export const REMOVE_ALERT = 'REMOVE_ALERT';
export const REGISTER_SUCCESS = 'REGISTER_SUCCESS';
export const REGISTER_FAIL = 'REGISTER_FAIL';
export const USER_LOADED = 'USER_LOADED';
export const AUTH_ERROR = 'AUTH_ERROR';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_FAIL = 'LOGIN_FAIL';
export const LOGOUT = 'LOGOUT';
export const GET_PROFILE = 'GET_PROFILE';
export const PROFILE_ERROR = 'PROFILE_ERROR';
export const CLEAR_PROFILE = 'CLEAR_PROFILE';
export const CREATE_PROFILE = 'CREATE_PROFILE';
export const UPDATE_PROFILE = 'UPDATE_PROFILE';
export const ACCOUNT_DELETED = 'ACCOUNT_DELETED';
export const GET_ALL_PROFILES = 'GET_ALL_PROFILES';
export const GET_PROFILE_BY_ID = 'GET_PROFILE_BY_ID';
export const GET_ALL_POSTS = 'GET_ALL_POSTS';
export const POSTS_ERROR = 'POSTS_ERROR';
export const CREATE_POST = 'CREATE_POST';
export const LIKE_A_POST = 'LIKE_A_POST';
export const UNLIKE_A_POST = 'UNLIKE_A_POST';
export const GET_POST = 'GET_POST';
export const COMMENT_ON_POST = 'COMMENT_ON_POST';
export const DELETE_POST = 'DELETE_POST';
export const DELETE_COMMENT = 'DELETE_COMMENT';
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Thu Oct 08 2020 14:52:20 GMT-0700 (Pacific Daylight Time)
            </div>
        </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    