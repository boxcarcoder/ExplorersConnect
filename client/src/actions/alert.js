import uuid from 'uuid';
import { SET_ALERT, REMOVE_ALERT } from './types';

// dispatch from thunk middleware
// the setAlert action will dispatch to the reducer
// an action consists of a type and may consist of a payload too
export const setAlert = (msg, alertType) => dispatch => {
  let id = uuid.v4();
  dispatch({
    type: SET_ALERT,
    payload: { msg, alertType, id }
  });
};
