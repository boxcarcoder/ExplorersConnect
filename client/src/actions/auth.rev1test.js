import mockAxios from 'axios';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { loadUser } from './auth';

// Create mock store that returns payloads as promises (thunk) and dispatches action statuses (promiseMiddleware).
const middlewares = [thunk, promiseMiddleware];
const mockStore = configureMockStore(middlewares);

describe('Auth Actions', () => {
  let store;

  // Initialize mock store before each test to prevent unexpected behavior.
  beforeEach(() => {
    store = mockStore({
      token: localStorage.getItem('token'),
      isAuthenticated: null,
      loading: true,
      loggedInUser: null,
    });
  });

  describe('loadUser action creator', () => {
    test('dispatches USER_LOADED action and returns data on success', async () => {
      // Mock the response of the action.
      mockAxios.get.mockImplementationOnce(() =>
        Promise.resolve({
          data: [
            { name: 'test name', email: 'test@test.com', avatar: '', date: '' },
          ],
        })
      );

      // Execute the test.
      await store.dispatch(loadUser());
      const actions = store.getActions();

      expect.assertions(2);
      expect(actions[0].type).toEqual('USER_LOADED');
      expect(actions[0].payload.data[0].name).toEqual('test name');

      // expect(actions[1].type).toEqual('USER_LOADED_FULFILLED');
      // expect(actions[1].payload.data[0].name).toEqual('test name');
      // expect(actions[1].payload.data[0].email).toEqual('test@test.com');
    });
  });
});
