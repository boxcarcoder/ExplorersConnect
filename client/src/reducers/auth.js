import {
  REGISTER_SUCCESS,
  REGISTER_FAIL,
  USER_LOADED,
  AUTH_ERROR,
  LOGIN_SUCCESS,
  LOGIN_FAIL,
  LOGOUT,
  ACCOUNT_DELETED
} from '../actions/types';

const initialState = {
  token: localStorage.getItem('token'), //fetches a token from local storage and assigns it to the token state
  isAuthenticated: null,
  loading: true,
  loggedInUser: null
};

export default function(state = initialState, action) {
  const { type, payload } = action;

  switch (type) {
    case USER_LOADED:
      return {
        ...state,
        isAuthenticated: true,
        loading: false,
        loggedInUser: payload //the payload is the user data from the backend from the GET request to /api/auth
      };

    case REGISTER_SUCCESS: // set the token into or remove from local storage
    case LOGIN_SUCCESS:
      localStorage.setItem('token', payload.token);
      return {
        ...state,
        ...payload,
        isAuthenticated: true,
        loading: false
      };

    case REGISTER_FAIL:
    case AUTH_ERROR:
    case LOGIN_FAIL:
    case LOGOUT:
    case ACCOUNT_DELETED:
      localStorage.removeItem('token');
      return {
        ...state,
        token: null, //set the token state to null since it is removed from local storage
        isAuthenticated: false,
        loading: false
      };

    default:
      return state;
  }
}
