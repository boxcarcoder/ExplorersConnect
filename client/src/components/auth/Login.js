import React, { useState } from 'react';
import { Link, Redirect } from 'react-router-dom';

//redux
import { connect } from 'react-redux';
import { login } from '../../actions/auth';
import PropTypes from 'prop-types';

const Login = ({ login, auth: { isAuthenticated } }) => {
  //each input requires a state and onChange handler
  const [formData, setFormData] = useState({
    email: '',
    password: ''
  });

  //destructuring: name === formData.name
  const { email, password } = formData;

  // spread operator: ...formData === copy of formData based on each attribute

  const handleEmailChange = e => {
    setFormData({ ...formData, email: e.target.value });
  };

  const handlePasswordChange = e => {
    setFormData({ ...formData, password: e.target.value });
  };

  //when the form is submitted, trigger the login action
  const handleSubmit = e => {
    e.preventDefault();
    login({ email, password });
  };

  // once the user is logged in, redirect to the dashboard
  if (isAuthenticated) {
    return <Redirect to='/dashboard' />;
  }

  return (
    <section className='container'>
      <h1 className='large text-primary'>Sign In</h1>
      <p className='lead'>
        <i className='fas fa-user-circle'> Start Exploring</i>
      </p>
      <form onSubmit={handleSubmit} className='form'>
        <div className='form-group'>
          <input
            type='email'
            placeholder='Email'
            value={email}
            onChange={e => handleEmailChange(e)}
            required
          />
        </div>
        <div className='form-group'>
          <input
            type='password'
            placeholder='Password'
            value={password}
            onChange={e => handlePasswordChange(e)}
            minLength='6'
          />
        </div>
        <input type='submit' value='Login' className='btn btn-primary' />
      </form>
      <p className='vert-m-1'>
        Don't have have an account? <Link to='/register'>Sign Up</Link>
      </p>
    </section>
  );
};

Login.propTypes = {
  login: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired
};

//mapping the auth redux state to a usable prop to check for authentication
const mapStateToProps = state => ({
  auth: state.auth //propName: state
});

export default connect(mapStateToProps, { login })(Login);
