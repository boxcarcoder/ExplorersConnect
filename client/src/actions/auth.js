import {
  REGISTER_SUCCESS,
  REGISTER_FAIL,
  USER_LOADED,
  AUTH_ERROR,
  LOGIN_SUCCESS,
  LOGIN_FAIL,
  LOGOUT,
  CLEAR_PROFILE,
} from './types';
import axios from 'axios';
import { setAlert } from './alert';
import setAuthToken from '../utils/setAuthToken';

// Load User.
// Sends an HTTP request to the backend to access an authenticated user's profile.
// Dispatches the user data returned from fetching into a redux state.
export const loadUser = () => async (dispatch) => {
  // set the token as the header if the token exists
  if (localStorage.token) {
    setAuthToken(localStorage.token);
  }

  try {
    // our backend returns the user as the res when fetching from /api/auth
    const res = await axios.get('/api/auth');

    // dispatch user data to reducer to save the user data into a redux state
    dispatch({
      type: USER_LOADED,
      payload: res.data,
    });
  } catch (err) {
    console.log('ERROR FROM AUTH ACTION:', err);
    dispatch({
      type: AUTH_ERROR,
    });
  }
};

// Register User.
// Sends an HTTP request to the backend to register the user.
// Dispatches the token returned from registering into a redux state.
export const register = ({ name, email, password }) => async (dispatch) => {
  // configuration of the HTTP request to the backend
  const config = {
    headers: {
      'Content-Type': 'application/json',
    },
  };

  // convert the object into a JSON string
  const body = JSON.stringify({ name, email, password });

  try {
    // our backend returns a token as the res when POSTing to api/users
    const res = await axios.post('/api/users', body, config);

    // dispatch the token to reducer to save the token into a redux state
    dispatch({
      type: REGISTER_SUCCESS,
      payload: res.data,
    });

    dispatch(loadUser());
  } catch (err) {
    // display errors as alerts using alert action and reducer
    // const errors = err.response.data.errors;
    // if (errors) {
    //   errors.forEach((error) => dispatch(setAlert(error.msg, 'danger')));
    // }
    console.log('err in auth action: ', err);
    // dispatch action to reducer to remove incorrect token from the redux state
    dispatch({
      type: REGISTER_FAIL,
    });
  }
};

// Login User.
// Sends an HTTP request to the backend to log in the user.
// Dispatches the token returned from logging in into a redux state.
export const login = ({ email, password }) => async (dispatch) => {
  // configuration of the HTTP request to the backend
  const config = {
    headers: {
      'Content-Type': 'application/json',
    },
  };

  // convert the object into a JSON string
  const body = JSON.stringify({ email, password });

  try {
    // our backend returns a token as the res when POSTing to api/auth
    const res = await axios.post('/api/auth', body, config);

    // dispatch the token to reducer to save the token into a redux state
    dispatch({
      type: LOGIN_SUCCESS,
      payload: res.data,
    });
    dispatch(loadUser());
  } catch (err) {
    // display errors as alerts using alert action and reducer
    const errors = err.response.data.errors;
    if (errors) {
      errors.forEach((error) => dispatch(setAlert(error.msg, 'danger')));
    }

    // dispatch action to reducer to remove incorrect token from the redux state
    dispatch({
      type: LOGIN_FAIL,
    });
  }
};

// Logout user and clear profile
export const logout = () => async (dispatch) => {
  dispatch({
    type: CLEAR_PROFILE,
  });
  dispatch({
    type: LOGOUT,
  });
};
