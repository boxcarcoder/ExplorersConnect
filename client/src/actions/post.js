import {
  GET_ALL_POSTS,
  POSTS_ERROR,
  SUBMIT_POST_SUCCESS,
  LIKE_A_POST
} from './types';
import axios from 'axios';
import setAuthToken from '../utils/setAuthToken';
import { setAlert } from './alert';

// Get all posts.
export const getAllPosts = () => async dispatch => {
  try {
    const res = await axios.get('/api/posts');

    dispatch({
      type: GET_ALL_POSTS,
      payload: res.data
    });
  } catch (err) {
    // dispatch error message and HTTP error status to the post redux state
    console.log('error getting all posts: ', err);
    dispatch({
      type: POSTS_ERROR,
      payload: { msg: err.response.statusText, status: err.response.status }
    });
  }
};

// Add a post.
export const addPost = formData => async dispatch => {
  try {
    //set the token as the header to gain access to the protected route POST /api/posts
    if (localStorage.token) {
      setAuthToken(localStorage.token);
    }

    // configuration of the HTTP request to the backend
    const config = {
      headers: {
        'Content-Type': 'application/json'
      }
    };

    console.log('attempting to place post into db.');
    const res = await axios.post('/api/posts', formData, config);

    dispatch({
      type: SUBMIT_POST_SUCCESS,
      payload: res.data
    });

    // display an alert to notify the user of what they just did
    dispatch(setAlert('Posted successfully.', 'success'));
  } catch (err) {
    console.log('error posting.');

    // dispatch error message and HTTP error status to the post redux state
    dispatch({
      type: POSTS_ERROR,
      payload: { msg: err.response.statusText, status: err.response.status }
    });
  }
};

// Like a post
export const likePost = id => async dispatch => {
  console.log('reached backend.');
  try {
    //set the token as the header to gain access to the protected route POST /api/posts/like
    if (localStorage.token) {
      setAuthToken(localStorage.token);
    }

    const res = await axios.put(`/like/${id}`);

    console.log('LIKE A POST: ', res);
    // dispatch({
    //   type: LIKE_A_POST,
    //   payload: res.data
    // });
  } catch (err) {
    console.log('error liking.');

    // dispatch error message and HTTP error status to the post redux state
    dispatch({
      type: POSTS_ERROR,
      payload: { msg: err.response.statusText, status: err.response.status }
    });
  }
};
