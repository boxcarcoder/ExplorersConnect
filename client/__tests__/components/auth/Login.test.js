import React from 'react';
import { shallow } from 'enzyme';

// Import the non-connected component for testing.
import { Login } from '../../../src/components/auth/Login';

// Props for the component.
import { login } from '../../../src/actions/auth';

//Globals for testing
let props;
let wrapper;

describe('<Login /> component.', () => {
  beforeEach(() => {
    props = {
      login: login,
      authState: {
        token: localStorage.getItem('token'),
        isAuthenticated: false,
        loading: false,
        loggedInUser: null,
      },
    };

    // Grab the shallow component in a wrapper.
    wrapper = shallow(<Login {...props} />);
  });

  test('Succesfully renders.', () => {
    expect(wrapper.exists()).toBe(true);
  });

  describe('Text box updates for', () => {
    test('Email.', () => {
      // Find the input within the component
      wrapper
        .find('input')
        .at(0)
        .simulate('change', {
          target: {
            value: 'test@test.com',
          },
        });

      // Execute the test
      expect(wrapper.find('input').at(0).props().value).toBe('test@test.com');
    });

    test('Password.', () => {
      // Find the input within the component
      wrapper
        .find('input')
        .at(1)
        .simulate('change', {
          target: {
            value: 'testpw',
          },
        });

      // Execute the test
      expect(wrapper.find('input').at(1).props().value).toBe('testpw');
    });
  });

  test('When the form is submitted, the default event is cancelled.', () => {
    // Find the form within the component to test it.
    let eventPrevented = false;
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {
        eventPrevented = true;
      },
    });
    // Execute the test.
    expect(eventPrevented).toBe(true);
  });

  test('Includes link to Register page.', () => {
    // Find the input within the component to test it.
    const link = wrapper.find('Link');

    // Execute the test
    expect(link.props().to).toBe('/register');
  });
});
