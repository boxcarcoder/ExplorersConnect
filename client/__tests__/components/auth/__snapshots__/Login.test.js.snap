// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> component. Successfully renders. 1`] = `
<section
  className="container"
>
  <h1
    className="large text-primary"
  >
    Sign In
  </h1>
  <p
    className="lead"
  >
    <i
      className="fas fa-user-circle"
    >
       Start Exploring
    </i>
  </p>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        minLength="6"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <input
      className="btn btn-primary"
      type="submit"
      value="Login"
    />
  </form>
  <p
    className="vert-m-1"
  >
    Don't have have an account? 
    <Link
      to="/register"
    >
      Sign Up
    </Link>
  </p>
</section>
`;
